primitives/runtime/src/traits.rs

```diff
@@ -636,6 +636,8 @@ pub type NumberFor<B> = <<B as Block>::Header as Header>::Number;
 pub type DigestFor<B> = Digest<<<B as Block>::Header as Header>::Hash>;
 /// Extract the digest item type for a block.
 pub type DigestItemFor<B> = DigestItem<<<B as Block>::Header as Header>::Hash>;
+/// Extract block hash.
+pub type BlockHashFor<B> = <B as Block>::Hash;
```

Comment:
I don't think this is necessary, in this case we're only doing one projection so the type will always be simple to write (just `B::Hash`).